src/layouts/ProfileLayout.tsx

```diff
@@ -45,69 +51,94 @@ function ProfileLayout({ children, wallet }: ProfileLayout): JSX.Element {
         />
         <link rel="icon" href="/favicon.ico" />
       </Head>
-      <Overview>
-        <Overview.Hero>
-          <Overview.Info
-            avatar={<Overview.Avatar src={wallet.previewImage} circle />}
-            title={<Overview.Title>{wallet.displayName}</Overview.Title>}
-          >
-            <Overview.Actions>
-              {isFollowingProfile ? (
+      <WalletIdentityProvider appName="Holaplex" appTwitter="@holaplex">
+        <Overview>
+          <Overview.Hero>
+            <Overview.Info
+              avatar={<Overview.Avatar src={wallet.previewImage} circle />}
+              title={<Overview.Title>{wallet.displayName}</Overview.Title>}
+            >
+              <Overview.Actions>
+                {isMyProfile && !wallet.profile?.handle && anchorWallet && (
```

Comment:
```suggestion
const { canConnect } = useConnectTwitter(address)
                {canConnect && (
```
Can read the viewer reactive var from apollo to get the if profile exists already and connected status. The viewer won't be available unless the user connects.

Reply:
But none but the viewer should be able to take action, right? Or even can? 

Maybe I'm misunderstanding...

Reply:
I think it's a bit simpler to not bring in a hook or the Apollo client here.
You get whether the wallet profile has handle directly from the page props, you need the anchor wallet as a prop to the component, and it should only be displayed on your own profile. So I vote it stays like this for now :)

Reply:
sounds good.